test/topology_custom/test_mv_admission_control.py

```diff
@@ -0,0 +1,132 @@
+                        "AND tablets = {'initial': 1}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Disable tablet balancing so that the slow node doesn't get tablets moved away from it.
+    for s in servers:
+        await manager.api.disable_tablet_balancing(s.ip_addr)
+    await manager.api.disable_tablet_balancing(server.ip_addr)
+
+    # Make sure that the slow node has a base table tablet and no view tablets, so that the
+    # view updates from it are remote. (using shard 0 and token 0 when moving tablets as they don't make a difference here)
+    base_tablet_replicas = await get_tablet_replicas(manager, servers[0], "ks", "tab", 0)
+    base_tablet_hosts = [str(replica[0]) for replica in base_tablet_replicas]
+    slow_host_id = await manager.get_host_id(server.server_id)
+    if str(slow_host_id) not in base_tablet_hosts:
+        base_tablet_host_id, base_tablet_shard = base_tablet_replicas[0]
+        await manager.api.move_tablet(servers[0].ip_addr, "ks", "tab", base_tablet_host_id, base_tablet_shard, slow_host_id, 0, 0)
+    view_tablet_replicas = await get_tablet_replicas(manager, servers[0], "ks", "mv_cf_view", 0)
+    view_tablet_hosts = [str(replica[0]) for replica in view_tablet_replicas]
+    for replica_host, replica_shard in view_tablet_replicas:
+        if str(replica_host) != str(slow_host_id):
+            continue
+        slow_host_shard = replica_shard
+        # Move the view tablet to the node that doesn't have one
+        for s in servers:
+            fast_host_id = await manager.get_host_id(s.server_id)
+            if str(fast_host_id) not in view_tablet_hosts:
+                await manager.api.move_tablet(servers[0].ip_addr, "ks", "mv_cf_view", slow_host_id, slow_host_shard, fast_host_id, 0, 0)
+        break
+
+    # Prepare the statement so that the write goes to the same shard
+    # for all requests (the backlog increase caused by a write is only
+    # immediately noted on the shard that the write was performed on).
+    stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+    for i in range(10):
+        # Perform a write that will increase the view update backlog on the slow node
+        # to a level causing admission control to reject the following writes.
+        await cql.run_async(stmt, [0, i, 240000*'a'], host=hosts[0])
+        # Based on whether the response from the slow node is received before the next write,
+        # the following small write can serve two purposes:
+        # 1. If the response is received before the next write, the write will be rejected by
+        #   admission control and retried until it reaches all replicas.
```

Comment:
Right, the driver does the retry. Also, because it's an `overloaded` exception, it looks like we actually keep retrying indefinitely as can be seen in https://github.com/scylladb/python-driver/blob/8a4387ae3f36522cc10841c607abb8182c6f8286/cassandra/policies.py#L870-L893

Reply:
Are you sure the driver retries indefinitely? As far as I know, in drivers "retry on next host" usually means "retry on the next node in the plan created by the load balancing policy" (plan == ordered list of replicas). I would guess that there will be at most as many retries as there are nodes in the cluster.

Reply:
That's what I observed. This was actually a problem in one of the tests I was trying along the way, when the server wouldn't stop replying with an 'overloaded' message and the test would never finish. Though I'm not sure if we can avoid that problem in the future - it can potentially occur for every `run_async` call when we're not using a custom retry policy